### ISSUE TITLE
Translations

### ISSUE THREAD

**Issue by RonnyAL** [yuhonas/free-exercise-db#9]
I'm thinking of utilizing your work in a project of mine - but I'd prefer to have texts in my own language. I'd be willing to do quite a bit of translation work myself, but it would be nice if translation could be a collaborative effort and not something done locally by individuals.

Would you be interested in implementing some kind of solution that supports this process?

**Comment by dev-horacebg** [comment]
I have the same requirement. Would be happy to contribute.

**Comment by yuhonas** [comment]
I like the idea, though few things to think about such file/folder/schema structure, tooling we'd need to track down missing translations etc anyone aware of some reference patterns/tooling/best practices we could adopt to streamline the process? I'll take a look around

**Comment by RonnyAL** [comment]
@yuhonas Thanks for getting back to us on this.

I don't currently have any input on specifically implementing this in your project, but I looked around trying to find a service that I've seen used by other FOSS projects. I think it's [POEditor](https://poeditor.com) - or at least this seems equally useful.

Since it might not be immediately apparent, the service offers free, unrestricted use for FOSS projects with an OSI-approved license:

From the [Pricing](https://poeditor.com/pricing/) page.

>If your software has an OSI-approved license, you can manage its localization completely free with POEditor. Just create a project and fill in the Open Source Project Request in its settings to get started.

Hoping this might give you some ideas on how to support translations.

Since you're here; thanks much for kindly providing this dataset :)

**Comment by yuhonas** [comment]
Great find, thanks! i've just submitted an FOSS application to POEditor, after that goes through i'll take a deeper look on how we can tackle it.

You're welcome, glad you're getting some use out of the project!

**Comment by 0x10-z** [comment]
I already translated to Spanish in my fork. If you are interested on it let me know

**Comment by diegods-ferreira** [comment]
> I already translated to Spanish in my fork. If you are interested on it let me know

I'm curious, how did you do it? Did you use a translating tool or manually? I'm insterested in translating to Portuguese (BR)...

**Comment by yuhonas** [comment]
> I already translated to Spanish in my fork. If you are interested on it let me know

Hi @0x10-z this one here https://github.com/yuhonas/free-exercise-db/pull/11 it looks like you're using Chat GPT to translate all the key/values then store them in a JSON "DB"? this could def be a great start

I'm assuming you're a native spanish speaker? what was the quality of the translations like?

**Comment by Programador-jr** [comment]
I would also really like to contribute to this, I made my version and a custom API and I would like to implement the function where it is possible to use multiple languages.

**Comment by yuhonas** [comment]
Yeah still keen! i think next steps (at least for me would be to agree on folder/file structure to support multiple languages (and lean on best practice/convention as much as we can), can you send a link to your version @Programador-jr interested in seeing it

**Comment by Programador-jr** [comment]
I haven't used my own translation yet because I didn't have enough time and it's a VERY large file. I imported the file into MongoDB because I thought it would be easier to implement a translation, I wanted to leave the files that way

```js
[
  {
    "_id": "6740c419026094d35227a20f",
    "name": {
      "en": "3/4 Sit-Up",
      "pt": "Abdominal 3/4"
    },
    "force": "pull",
    "level": "beginner",
    "mechanic": "compound",
    "equipment": {
      "en": "body only",
      "pt": "apenas o corpo"
    },
    "primaryMuscles": {
      "en": ["abdominals"],
      "pt": ["abdominais"]
    },
    "secondaryMuscles": {
      "en": [],
      "pt": []
    },
    "instructions": {
      "en": [
        "Lie down on the floor and secure your feet. Your legs should be bent at the knees.",
        "Place your hands behind or to the side of your head. You will begin with your back on the ground. This will be your starting position.",
        "Flex your hips and spine to raise your torso toward your knees.",
        "At the top of the contraction your torso should be perpendicular to the ground. Reverse the motion, going only ¾ of the way down.",
        "Repeat for the recommended amount of repetitions."
      ],
      "pt": [
        "Deite-se no chão e segure seus pés. Suas pernas devem estar dobradas nos joelhos.",
        "Coloque suas mãos atrás ou ao lado da cabeça. Você começará com as costas no chão. Esta será sua posição inicial.",
        "Flexione seus quadris e coluna para levantar o tronco em direção aos joelhos.",
        "No topo da contração, seu tronco deve estar perpendicular ao chão. Inverta o movimento, descendo apenas ¾ do caminho.",
        "Repita para a quantidade recomendada de repetições."
      ]
    },
    "category": {
      "en": "strength",
      "pt": "força"
    },
    "images": [
      "3_4_Sit-Up/0.jpg",
      "3_4_Sit-Up/1.jpg"
    ],
    "id": "3_4_Sit-Up",
    "__v": 0
  }
]
```
